internal/source/keptn/source.go

```diff
@@ -0,0 +1,147 @@
+package keptn
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kubeshop/botkube/internal/loggerx"
+	"github.com/kubeshop/botkube/pkg/api"
+	"github.com/kubeshop/botkube/pkg/api/source"
+)
+
+const (
+	// PluginName is the name of the Keptn Botkube plugin.
+	PluginName = "keptn"
+
+	description = "Keptn plugin polls events from configured Keptn API endpoint."
+
+	pollPeriodInSeconds = 5
+)
+
+// Source prometheus source plugin data structure
+type Source struct {
+	pluginVersion string
+}
+
+// NewSource returns a new instance of Source.
+func NewSource(version string) *Source {
+	return &Source{
+		pluginVersion: version,
+	}
+}
+
+// Stream streams Keptn events
+func (p *Source) Stream(ctx context.Context, input source.StreamInput) (source.StreamOutput, error) {
+	out := source.StreamOutput{Event: make(chan source.Event)}
+	config, err := MergeConfigs(input.Configs)
+	if err != nil {
+		return source.StreamOutput{}, fmt.Errorf("while merging input configs: %w", err)
+	}
+	go p.consumeEvents(ctx, config, out.Event)
+
+	return out, nil
+}
+
+// Metadata returns metadata of Keptn configuration
+func (p *Source) Metadata(_ context.Context) (api.MetadataOutput, error) {
+	return api.MetadataOutput{
+		Version:     p.pluginVersion,
+		Description: description,
+		JSONSchema:  jsonSchema(),
+	}, nil
+}
+
+func (p *Source) consumeEvents(ctx context.Context, cfg Config, ch chan<- source.Event) {
+	keptn, err := NewClient(cfg.URL, cfg.Token)
+	log := loggerx.New(cfg.Log)
+	exitOnError(err, log)
+
+	for {
```

Comment:
I think we should exit the for loop in case `ctx.Done()`. Please update this. Thank you!